service/controller/resource/capzcrs/create.go

```diff
@@ -115,12 +116,13 @@ func (r *Resource) mapAzureConfigToCluster(ctx context.Context, cr providerv1alp
 			Namespace: key.OrganizationNamespace(&cr),
 			Labels: map[string]string{
 				// XXX: azure-operator reconciles Cluster & MachinePool to set OwnerReferences (for now).
-				label.AzureOperatorVersion:    key.OperatorVersion(&cr),
-				label.ClusterOperatorVersion:  cr.Labels[label.ClusterOperatorVersion],
-				label.Cluster:                 cr.Name,
-				capiv1alpha3.ClusterLabelName: cr.Name,
-				label.Organization:            key.OrganizationID(&cr),
-				label.ReleaseVersion:          key.ReleaseVersion(&cr),
+				label.AzureOperatorVersion:          key.OperatorVersion(&cr),
+				label.ClusterOperatorVersion:        cr.Labels[label.ClusterOperatorVersion],
+				label.Cluster:                       cr.Name,
+				capiv1alpha3.ClusterLabelName:       cr.Name,
+				label.Organization:                  key.OrganizationID(&cr),
+				label.ReleaseVersion:                key.ReleaseVersion(&cr),
+				azopannotation.UpgradingToNodePools: "True",
```

Comment:
New annotation to signal that we are upgrading to node pools release for the first time. It is removed as soon as the upgrade is completed.